# --- ROCK-COPYRIGHT-NOTE-BEGIN ---
# 
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
# Please add additional copyright information _after_ the line containing
# the ROCK-COPYRIGHT-NOTE-END tag. Otherwise it might get removed by
# the ./scripts/Create-CopyPatch script. Do not edit this copyright text!
# 
# ROCK Linux: rock-src/package/*/ploticus/axis-gridblocks.patch
# Copyright (C) 1998 - 2004 ROCK Linux Project
# 
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# 
# --- ROCK-COPYRIGHT-NOTE-END ---

This fixes (workarounds) gridblock drawing. The original code drew the
whole area, and then started at the top overdrawing the first block and
thus missing one at the button (or reversed if reversed ...).

This changes the behaviour to always draw downwards (or reversed) and thus
at least yielding the correct gridblocks - although the performance is
bad (many block fills ...).

  - Rene Rebe <rene@rocklinux.org>

diff -ur pl221src/src/proc_axis.c pl221src-fixed/src/proc_axis.c
--- pl221src/src/proc_axis.c	2004-05-11 21:11:53.000000000 +0200
+++ pl221src-fixed/src/proc_axis.c	2004-06-04 18:50:19.000000000 +0200
@@ -937,7 +937,7 @@
 				Ecblock( glemin, gbylast, glemax, Ea( Y, y ), gbcolor2, 0 );
 				gbstate = 0;
 				}
-			gbylast = Ea( Y, y );
+			/* gbylast = Ea( Y, y ); */
 			}
 		else	{
 			linedet( "grid", grid, 0.5 );
